tests/e2e/vsphere_shared_datastore.go

```diff
@@ -121,4 +126,162 @@ var _ = ginkgo.Describe("[csi-block-vanilla] [csi-block-vanilla-parallelized] "+
 		gomega.Expect(errorOccurred).To(gomega.BeTrue())
 
 	})
+
+	// Verify impact on existing pv pvc when sc recreated with different binding mode
+	// Steps
+	// 1. Create a Storage Class with Immediate Binding Mode.
+	// 2. Create a PVC using above SC.
+	// 3. Wait for PVC to be in Bound phase.
+	// 4. Create standalone pod
+	// 5. Delete Storgae Class created above
+	// 6. Recreate Storage Class with same name as above but with Binding mode WaitForFirstConsumer
+	// 7. Check PVC status after recreating storage class
+	// 8. Check pod status after recreating storage class
+	// 9. Delete Pod
+	// 10. Delete PVC and SC
+
+	ginkgo.It("[csi-block-vanilla] [csi-guest] [csi-supervisor] "+
+		"Verify impact on existing pv pvc when sc recreated with different binding mode", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Invoking Test to verify impact on existing pv pvc when sc recreated with different binding mode")
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+		if !vanillaCluster {
+			storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores2)
+		}
+		// Create Storage class and PVC
+		ginkgo.By("Creating Storage Class and PVC")
+		if guestCluster {
+			scParameters[svStorageClassName] = storagePolicyName
+			storageclass, pvclaim, err = createPVCAndStorageClass(client,
+				namespace, nil, scParameters, diskSize, nil, "", false, "")
+		} else if supervisorCluster {
+			namespace = getNamespaceToRunTests(f)
+			profileID := e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+			scParameters[scParamStoragePolicyID] = profileID
+			createResourceQuota(client, namespace, rqLimit, storagePolicyName)
+			storageclass, pvclaim, err = createPVCAndStorageClass(client,
+				namespace, nil, scParameters, diskSize, nil, "", false, "", storagePolicyName)
+		} else if vanillaCluster {
+			scParameters[scParamFsType] = ext4FSType
+			storageclass, pvclaim, err = createPVCAndStorageClass(client,
+				namespace, nil, scParameters, "", nil, "", false, "")
+		}
+		framework.Logf("storageclass name :%s", storageclass.GetName())
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			if !supervisorCluster {
```

Comment:
since we are using shared-ds-policy2 in this test , we can delete this storage class. This  supervisorCluster check is not needed